vnext/Microsoft.ReactNative/Views/ScrollViewManager.cpp

```diff
@@ -121,6 +131,71 @@ void ScrollViewShadowNode::createView() {
       });
 }
 
+void ScrollViewShadowNode::OnPreviewKeyDown(
+    const winrt::IInspectable &sender,
+    const xaml::Input::KeyRoutedEventArgs &e) {
+  auto senderFrameworkElement = sender.try_as<winrt::FrameworkElement>();
+  auto xamlRoot = senderFrameworkElement.XamlRoot();
+  auto focusedElement =
+      xaml::Input::FocusManager::GetFocusedElement(xamlRoot).try_as<winrt::Windows::UI::Xaml::DependencyObject>();
+  auto findNextElementOptions = xaml::Input::FindNextElementOptions();
+  findNextElementOptions.SearchRoot(xamlRoot.Content());
+  auto previousKey = m_isHorizontal ? winrt::Windows::System::VirtualKey::Left : winrt::Windows::System::VirtualKey::Up;
+  auto nextKey = m_isHorizontal ? winrt::Windows::System::VirtualKey::Right : winrt::Windows::System::VirtualKey::Down;
+
+  if (e.Key() == previousKey) {
+    auto firstFocusableElement =
+        xaml::Input::FocusManager::FindFirstFocusableElement(senderFrameworkElement).try_as<xaml::DependencyObject>();
+    auto isFirstElementFocused = firstFocusableElement == focusedElement;
+    if (!isFirstElementFocused) {
+      xaml::Input::FocusManager::TryMoveFocus(xaml::Input::FocusNavigationDirection::Previous, findNextElementOptions);
+      e.Handled(true);
+    }
+  } else if (e.Key() == nextKey) {
+    auto lastFocusableElement =
+        xaml::Input::FocusManager::FindLastFocusableElement(senderFrameworkElement).try_as<winrt::DependencyObject>();
+    auto isLastElementFocused = lastFocusableElement == focusedElement;
+    if (!isLastElementFocused) {
+      xaml::Input::FocusManager::TryMoveFocus(xaml::Input::FocusNavigationDirection::Next, findNextElementOptions);
+      e.Handled(true);
+    }
+  } else if (e.Key() == winrt::Windows::System::VirtualKey::Tab) {
+    auto const &coreWindow = winrt::CoreWindow::GetForCurrentThread();
+    auto isShiftDown = KeyboardHelper::IsModifiedKeyPressed(coreWindow, winrt::Windows::System::VirtualKey::Shift);
+
+    winrt::Point anchorTopLeft = winrt::Point(0, 0);
+    xaml::Media::GeneralTransform transform = senderFrameworkElement.TransformToVisual(xamlRoot.Content());
+    winrt::Point anchorTopLeftConverted = transform.TransformPoint(anchorTopLeft);
+    auto exclusionRect = winrt::Rect(
+        anchorTopLeftConverted.X,
+        anchorTopLeftConverted.Y,
+        static_cast<float>(senderFrameworkElement.Width()),
+        static_cast<float>(senderFrameworkElement.Height()));
```

Comment:
I found a bug here: We should be using `ActualWidth` and `ActualHeight`

Reply:
What difference have you noticed it made?

Reply:
In cases where the FocusZoneView didn't have an explicit width/height it wouldn't return the correct next element. The Width/Height properties are set only if you explicitly size a control in Xaml. You should always use the `Actual` props because they will give you the computed size of a control. @asklar helped me debug this.